src/main/java/org/veupathdb/eda/binaryfiles/printer/BinaryFilePrinter.java

```diff
@@ -117,20 +120,22 @@ private static void printIdsMapFile(Path binaryFile, JSONObject metajson) {
     }
   }    
   
-  private static void printVarFile(Path binaryFile, JSONObject metajson) {
+  private static void printVarFile(Path binaryFile, BinaryFilesManager.Metadata metadata) throws Exception {
+    BinaryFilesManager.VariableMeta variableMeta = metadata.getVariableMetadata().stream()
+        .filter(var -> binaryFile.getFileName().toString().contains(var.getVariableId()))
+        .findFirst()
+        .orElseThrow();
```

Comment:
Add custom exception here with message of what went wrong- is this always expected?